### PR TITLE
Add self force turn on function

### PR DESCRIPTION
**Pull request by nikwit**
Adds the turn on function used to continuously turn on the self force.

### CODE REVIEW

File: src/Evolution/Systems/CurvedScalarWave/Worldtube/SelfForce.hpp [hunk 1]
```diff
@@ -120,4 +125,29 @@ tnsr::A<double, Dim> dt_Du_self_force_per_mass(
...
+
+/*!
+ * \brief A function used to roll-on the self-force continuously from 0 to 1
+ *
+ * \details It is given by Eq.(60) of Wittek:2024gxn
+ * \begin{equation}
+ * w(t) = 1 - \exp \left( - \left(t / \sigma \right)^4 \right),
+ * \end{equation}
+ * where $t$ is the time where the self-force is turned on and $\sigma$ dictates
+ * the timescale over which it is turned on. The function is $\mathcal{C}^3$,
```
Comment (knelli2) [inline src/Evolution/Systems/CurvedScalarWave/Worldtube/SelfForce.hpp]: Wait is $t$ a single time, or is it just the current time? Seems like it's $t = t_{\mathrm{cur}} - t_{\mathrm{ON}}$. Update somehow to make this clearer.

File: src/Evolution/Systems/CurvedScalarWave/Worldtube/SelfForce.cpp [hunk 2]
```diff
@@ -133,6 +147,29 @@ tnsr::A<double, Dim> dt_Du_self_force_per_mass(
...
+  detail::check_positive_time_and_timescale(time, turn_on_timescale);
+  const double t_over_timescale = time / turn_on_timescale;
+  return 1. - exp(-square(square(t_over_timescale)));
+}
+
+double dt_turn_on_function(const double time, const double turn_on_timescale) {
+  detail::check_positive_time_and_timescale(time, turn_on_timescale);
+  const double t_over_timescale = time / turn_on_timescale;
+  return 4. * cube(time) * exp(-square(square(t_over_timescale))) /
+         square(square(turn_on_timescale));
```
Comment (knelli2) [inline src/Evolution/Systems/CurvedScalarWave/Worldtube/SelfForce.cpp]: [optional] Less operations
```cpp
  const double t_over_timescale_quart = square(square(time / turn_on_timescale));
  return 4. * t_over_timescale_quart * exp(-t_over_timescale_quart) / time;
```

**Review by knelli2 — CHANGES_REQUESTED** [review]
These and clang-tidy. You can squash directly

**Comment by nikwit** [comment]
done. Thanks for the review!

**Review by knelli2 — APPROVED** [review]

**Pull request merged**